File: app/src/main/java/com/mandarin/bcu/androidutil/adapters/SingleClick.java
package com.mandarin.bcu.androidutil.adapters;

import android.os.SystemClock;
import android.view.View;

public abstract class SingleClick implements View.OnClickListener {
    private static final long INTERVAL = 1000;

    private long mLastClickTime;

    public abstract void onSingleClick(View v);

    @Override
    public final void onClick(View v) {
        long currentTime = SystemClock.uptimeMillis();
        long elapsed = currentTime - mLastClickTime;
        mLastClickTime = currentTime;

        if (elapsed <= INTERVAL) {
            return;
        }

        onSingleClick(v);
    }
}

File: app/src/main/java/com/mandarin/bcu/androidutil/adapters/SingleClick.kt
package com.mandarin.bcu.androidutil.adapters

import android.os.SystemClock
import android.view.View

abstract class SingleClick : View.OnClickListener {
    private var mLastClickTime: Long = 0
    abstract fun onSingleClick(v: View?)
    override fun onClick(v: View) {
        val currentTime = SystemClock.uptimeMillis()
        val elapsed = currentTime - mLastClickTime
        if (elapsed <= INTERVAL) {
            return
        }

        mLastClickTime = currentTime
        onSingleClick(v)
    }

    companion object {
        private const val INTERVAL: Long = 1000
    }
}